src/Core/Managed/Shared/Channel/TelemetryBuffer.cs

```diff
@@ -46,7 +48,30 @@ public int Capacity
                 this.capacity = value;
             }
         }
-        
+
+        /// <summary>
+        /// Gets or sets the maximum number of telemetry items that can be in the backlog to send. Items will be dropped
+        /// once this limit is hit.
+        /// </summary>        
+        public int MaximumUnsentBacklogSize
+        {
+            get
+            {
+                return this.maximumUnsentBacklogSize;
+            }
+
+            set
+            {
+                if (value < 1)
+                {
+                    this.maximumUnsentBacklogSize = DefaultMaximumUnsentBacklogSize;
```

Comment:
Yes but in this case the behavior is worse than the principle (and I already disagree with the principle).  I agree with not breaking existing customers, at least without increasing the major version number, but this is net new and won't break anyone so the behavior of ignoring their setting and not telling them is clearly a bug/broken/bad design.